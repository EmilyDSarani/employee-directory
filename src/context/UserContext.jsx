import { useContext, createContext, useState } from 'react'
import { getUser } from '../services/user';

//We will use this to set the user in context and have it be across the entire app.
//it is a big bucket of state that is shared across the entire app

//Create Context
const UserContext = createContext();

//Create Provider
const UserProvider = ({ children }) => {
    const currentUser = getUser()
    //when I load any page, its going to try to get the user from local storage
    //if that user exists, then set the email to user email
    const [user, setUser] = useState(currentUser?{email:currentUser.email} : {})

    //userAuth
    const login = (email, password) =>{
        const loginSuccessful =
        email === '' &&
        password === '';
        if(loginSuccessful) setUser(email);
        return loginSuccessful
    }
    //set logout
    const logout =(callback) =>{
        setUser(null);
        callback();
    };
    return <UserContext.Provider value={{user, setUser, login, logout}}>{children}</UserContext.Provider>
}
    //useUser Custom Hook
    const useUser = () => {
        const context = useContext(UserContext)

        if(context === undefined){
            throw new Error('useUser needs to be defined in UserContext Provider')
        }
        return context
    }


export {useUser, UserProvider}