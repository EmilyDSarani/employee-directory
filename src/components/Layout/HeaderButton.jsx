import { useUser } from "../../context/UserContext";
import { Link, useHistory } from "react-router-dom"
import { signOutUser } from "../../services/user";

export default function HeaderButton() {
    const { user, setUser } = useUser();
    const history = useHistory();

    const handleLogOut = async () => {
        await signOutUser();
        setUser({})
        history.push('/login');
    };

    return (
        <div>
              {user?.email ? (
                  <button onClick={handleLogOut}>LogOut</button>
              ) : (
            <Link to="/login"><button>Sign in</button></Link>
              )}
        </div>
    )
}
