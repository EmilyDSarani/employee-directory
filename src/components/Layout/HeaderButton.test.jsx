import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router'
import { UserProvider } from '../../context/UserContext'
import HeaderButton from './HeaderButton'

jest.mock('../../context/UserContext.jsx')
jest.mock('../../services/user.js')

it.skip('should display login button', ()=> {
    const { container } = render(
        <UserProvider>
            <MemoryRouter>
                <HeaderButton />
            </MemoryRouter>
        </UserProvider>
    );
    expect(container).toMatchInlineSnapshot(`
       <div>
         <a
           href="/login"
         >
           <button>
             Sign in
           </button>
         </a>
       </div>
     `)
});