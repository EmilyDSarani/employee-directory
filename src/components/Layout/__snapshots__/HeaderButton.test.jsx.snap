// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display login button 1`] = `
<div>
  <div>
    <a
      href="/login"
    >
      <button>
        Sign in
      </button>
    </a>
  </div>
</div>
`;
